argus/cbinit/base_config_generator.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2016 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import abc
+import ConfigParser
+import ntpath
+import StringIO
+
+from argus import util
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseCBInitConfig(object):
```

Comment:
A better hierarchy can be:
 - BaseConfig(object)
 - CloudbaseInitConfig(BaseConfig)